Projet-Web-Avance-A4/Developpeur
# Add modify delete crud + Tokens +style

Ticket : [Trello](https://trello.com/c/MI9Ah44r/55-crud-développeur)

**How to test ?**

1) Checkout to this branch (same for [Microservices](https://github.com/Projet-Web-Avance-A4/Microservices/compare/dev...add-modify-delete-crud-developpeur?expand=1))

2) `npm run dev` on Developpeur (1st) and Microservices (2nd)

3) Create an account, if successful, you will be redirected to login component after 1 second.

4) Connect to your account

5) Check your localStorage if both tokens (accessToken & refreshToken) were set.

![image](https://github.com/Projet-Web-Avance-A4/Developpeur/assets/141679077/a2cacc93-f3e0-4fb8-8efe-be589f4407e2)

6) On download page, cards should be in a grid with 3 Cards a row.

7) Go to Your account, your data should be shown like this : 

![image](https://github.com/Projet-Web-Avance-A4/Developpeur/assets/141679077/41a194f4-3f6c-4978-8bcf-17b793169c90)

8) Try updating some information, tokens will be updating, and the new data will be shown. (Exemple : street was changed)

![image](https://github.com/Projet-Web-Avance-A4/Developpeur/assets/141679077/cc77c776-6184-42cf-abf3-104e533b87e2)

9) Try updating your password, then go back to login page by clicking on the logo in the header, and connect with your new password. Old password will not work anymore.

10) Try deleting your account, if your password is correct, you will be redirected to login page. If you try to connect, it will not work.

11) If you recreate you account with the same mail, it will works (status become "Active" instead of "Deleted"). If you entered new data, it will update it in the database.
